### PR TITLE
notification model added as well as routes involved

### PR DESCRIPTION
**Pull request by christian-barnabe-chabi**
### What this PR does (required):
- create notification model and routes involved

### Screenshots / Videos (required):
- Post at least one screenshot to show the feature in action

### Any information needed to test this feature (required):
- Post what steps to take to be able to test this feature (eg. "sign up as a new user and upload a profile picture from the navbar")

### Any issues with the current functionality (optional):
- List any problems where you were not able to complete all tasks on ticket
- List any problems this PR may cause for the project or other active PRs
- Post a screenshot/video of the problem, along with detailed console outputs where applicable

### CODE REVIEW

File: server/models/Notification.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  },
+  new_price: {
+    type: Number,
+    required: true,
+  },
+  read: {
+    type: Boolean,
+    required: true,
+    default: false,
+  },
```
Comment (sina-jamshidi) [inline server/models/Notification.js:34]: if we have different types of notifications (other than product price dropping), then these should probably not be required

Comment (christian-barnabe-chabi) [reply server/models/Notification.js:34]: got it. thank you @sina-jamshidi

File: server/routes/notifications.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+const express = require('express');
+const protect = require('../middleware/auth');
+const router = express.Router();
+
+// get all
+router.route('/').get(protect, (req, res)=>res.send());
```
[the remark this answers is not in the document]
Comment (christian-barnabe-chabi) [reply server/routes/notifications.js]: fixed

File: server/controllers/notification.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        success: {
+          message: "Notification updated",
+        }
+      });
+    }
+  });
+});
+
+// i'm not sure this is useful as notifications are going to be created on some internal events
+exports.create = expressAsyncHandler((req, res, next) => {
```
Comment (sina-jamshidi) [inline server/controllers/notification.js:88]: I see your point. It's true we are going to be creating notifications in other controllers. We can keep this here for now though

**Review by sina-jamshidi — APPROVED** [review]
Looks good to go!

**Pull request merged**